### PR TITLE
bind: Add JSON parameter to DescribeFlags

### PR DESCRIPTION
**Pull request by alexh-sauce**
This PR adds an option to describe flags in JSON format.

This is useful for exporting or recording config settings.

### CODE REVIEW

File: bind/flag.go [hunk 1]
```diff
@@ -25,6 +27,13 @@ import (
 	"github.com/spf13/pflag"
 )
 
+type Format int
```
Comment (Choraden) [inline bind/flag.go:30]: Let's name it `DescribeFormat`. This is `bind` package and `bind.Format` might be misleading.

File: bind/flag.go [hunk 2]
```diff
@@ -25,6 +27,13 @@ import (
...
+
+const (
+	Plain Format = 0
+	JSON  Format = 1
+)
```
Comment (Choraden) [inline bind/flag.go:35]: nit: 
```go
const (
	Plain Format = iota
	JSON
)
```

File: bind/flag.go [hunk 3]
```diff
@@ -328,13 +337,41 @@ func MarkFlagFilename(cmd *cobra.Command, names ...string) {
...
+		if flag.Name == "help" {
+			return
+		}
+
+		if flag.Hidden && !showHidden {
 			return
 		}
-		b.WriteString(fmt.Sprintf("%s=%s\n", flag.Name, strings.Trim(flag.Value.String(), "[]")))
+
+		if flag.Value.Type() == "bool" {
```
Comment (Choraden) [inline bind/flag.go:353]: I don't get this case. Please explain.

Comment (Choraden) [reply bind/flag.go:353]: ah we don't want the booleans to be quoted.

File: bind/flag.go [hunk 4]
```diff
@@ -328,13 +337,41 @@ func MarkFlagFilename(cmd *cobra.Command, names ...string) {
 	}
 }
 
-func DescribeFlags(fs *pflag.FlagSet) string {
-	var b strings.Builder
+func DescribeFlags(fs *pflag.FlagSet, showHidden bool, format Format) (string, error) {
+	args := make(map[string]any, 0)
+	keys := make([]string, 0)
```
Comment (Choraden) [inline bind/flag.go:342]: nit: use fs.NFlag() to preallocate

File: cmd/forwarder/run/run.go [hunk 5]
```diff
@@ -43,7 +43,10 @@ type command struct {
 }
 
 func (c *command) RunE(cmd *cobra.Command, _ []string) error {
-	config := bind.DescribeFlags(cmd.Flags())
+	config, err := bind.DescribeFlags(cmd.Flags(), false, bind.JSON)
```
Comment (Choraden) [inline cmd/forwarder/run/run.go:46]: How about we stick to the Plain format here?

**Comment by Choraden** [comment]
How about we move it to a separate file e.g. `describe.go`?

File: cmd/forwarder/run/run.go (hunk 5 above)
Comment (alexh-sauce) [reply cmd/forwarder/run/run.go]: Oof, yes. Thanks for catching that.

**Comment by alexh-sauce** [comment]
@Choraden Make sense, I've moved the function to a new file.

File: bind/describe.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+type DescribeFormat int
+
+const (
+	Plain DescribeFormat = iota
+	JSON
+)
+
+func DescribeFlags(fs *pflag.FlagSet, showHidden bool, format DescribeFormat) (string, error) {
+	args := make(map[string]any, fs.NFlag())
+	keys := make([]string, fs.NFlag())
```
[the remark this answers is not in the document]
Comment (alexh-sauce) [reply bind/describe.go:27]: Ah, gotcha.

**Review by Choraden — APPROVED** [review]
LGTM

**Pull request merged**